ocaml/typing/persistent_env.ml

```diff
@@ -74,20 +74,37 @@ type can_load_cmis =
   | Can_load_cmis
   | Cannot_load_cmis of Lazy_backtrack.log
 
-type pers_struct = {
-  ps_is_param: bool;
-  ps_crcs: Import_info.t array;
-  ps_filename: string;
-  ps_visibility: Load_path.visibility;
+(* Data relating directly to a .cmi *)
+type import = {
+  imp_is_param : bool;
+  imp_impl : Impl.t;
+  imp_sign : Subst.Lazy.signature;
+  imp_filename : string;
+  imp_visibility: Load_path.visibility;
+  imp_crcs : Import_info.Intf.t array;
+  imp_flags : Cmi_format.pers_flags list;
 }
 
 (* If a .cmi file is missing (or invalid), we
    store it as Missing in the cache. *)
-type 'a pers_struct_info =
+type import_info =
   | Missing
-  | Found of pers_struct * 'a
+  | Found of import
+
+type binding =
+  | Static of Compilation_unit.t (* Bound to a static constant *)
+
+(* Data relating to an actual referenceable module, with a signature and a
+   representation in memory. *)
+type pers_struct = {
+  ps_import: import;
+  ps_binding: binding;
```

Comment:
This field is not actually read anywhere. Presumably it's used in later PRs? Would probably have been better to add it in when it was used.

Reply:
Actually, I took out both fields for the time being. The pointer through to the `import` might not actually be useful after all either.